ulib/Makefile

```diff
@@ -6,22 +6,33 @@ FSTAR_HOME=..
 # Makefile that is distributed in the binary package. This
 # Makefile here is not.
 
-all:
+all: .cache
 	+$(MAKE) FSTAR_HOME=$(FSTAR_HOME) -f Makefile.verify verify-core
 
-benchmark:
+benchmark: .cache
 	+$(MAKE) FSTAR_HOME=$(FSTAR_HOME) -f Makefile.verify verify-benchmark
 
-extra:
+extra: .cache
 	+$(MAKE) FSTAR_HOME=$(FSTAR_HOME) -f Makefile.verify verify-extra
 
+.cache:
+	mkdir -p .cache
+
 include ml/Makefile.include
 
 REALIZED=$(addsuffix .fst, $(NOEXTRACT_MODULES))
 
-FSTARLIB_MODULES=$(filter-out $(REALIZED) FStar.Crypto.fst FStar.Tcp.fst FStar.Reflection.% FStar.Tactics.%,$(wildcard FStar.*.fst FStar.*.fsti)) $(wildcard LowStar.*.fst LowStar.*.fsti)
-
-FSTAR_TAC_LIB_MODULES=$(filter-out $(REALIZED),$(wildcard FStar.Reflection.*.fst FStar.Reflection.*.fsti FStar.Tactics.*.fst FStar.Tactics.*.fsti))
+FSTAR_LIB_EXCLUSIONS=$(REALIZED) FStar.Crypto.fst FStar.Tcp.fst FStar.Reflection.% FStar.Tactics.%
+FSTAR_LIB_ALL_MODULES=\
+	$(wildcard FStar.*.fst FStar.*.fsti) \
+	$(wildcard LowStar.*.fst LowStar.*.fsti) \
+	$(wildcard legacy/*fst legacy/*fsti) \
+	$(wildcard experimental/*fst experimental/*fsti)
+FSTARLIB_MODULES=$(filter-out $(FSTAR_LIB_EXCLUSIONS), $(FSTAR_LIB_ALL_MODULES))
```

Comment:
I'm not sure this still works (or if it ever did); consider FStar.Buffer which appears in NOEXTRACT_MODULES -- it's now in legacy, so because it appears without the `legacy/` prefix, it can't be filtered-out

However, it already had a `-` prefix so maybe that filtering-out never worked. Not sure what's going on here.

Reply:
Note that `FStar.Crypto` is no longer filtered-out for sure (leaving the question about NOEXTRACT_MODULES aside)

Reply:
Thanks for noticing this. 

It looks like this can be simplified a lot. What matters is the value of EXTRACT_MODULES when calling Makefile.extract. The FSTAR_FILES variable is only used to set the roots for dependence scanning. I'll revise it now

Reply:
I'm not sure about why FStar.Crypto is being filtered out. There isn't a corresponding FStar_Crypto.ml, so I think it should be retained, even though it doesn't contain much.